test/compilable/cppmangle.d

```diff
@@ -378,3 +378,19 @@ else version (Posix)
     static assert(T.boo.mangleof == "_ZN1T3booE");
 }
 
+/****************************************/
+// https://github.com/dlang/dmd/pull/6658
+
+
+extern (C++, std)
+{
+    struct pair(T1, T2)
+    {
+	void swap(ref pair other);
+    }
+}
+
+version (linux)
+{
+    static assert(std.pair!(void*, void*).swap.mangleof == "_ZNSt4pairIPvS0_E4swapERS1_");
```

Comment:
Finally, what changes is actually related to this fix? All I can see are refactorings.

Reply:
I added a bug report for the fix, and commented the code where the two corrections are.

Reply:
Thanks, I see it.